pagopa/io-app
# chore: [IOPID-670,IOPID-714] Fix EIC login flow + EIC dev server login local FF

## Short description
This PR fixes a bug occurring during the EIC login flow: the user after settings their email address stare an endless spinner and cannot do anything else but restarting the app.

In the meanwhile, this PR add a new local FF (`CIE_LOGIN_WITH_DEV_SERVER_ENABLED`) that allow to start a login flow from the EIC login button in the landing page.

| Before | After | 
| - | - |
| <video src="https://github.com/pagopa/io-app/assets/16268789/c71941f8-b9e6-4054-8ebd-20ef7501733d" /> | <video src="https://github.com/pagopa/io-app/assets/16268789/be1843c2-3658-4b01-b3dd-78d18c2f776b" />  |

## How to test
 - **Avoid Regression**: build in production mode and test an EIC successful login.
 - **EIC Login with dev server**: set `CIE_LOGIN_WITH_DEV_SERVER_ENABLED=YES` and on the dev server config:
 ```ts
profile: {
    attrs: defaultProfileAttrs,
    authenticationProvider: "cie",
    firstOnboarding: true,
    allowRandomValues: true
},
```
Then, after the login, you should be able to set your email and go through the remaining steps of your first onboarding.
